row.go

```diff
@@ -739,10 +739,21 @@ func reconstructFuncOfGroup(columnIndex int16, node Node) (int16, reconstructFun
 
 //go:noinline
 func reconstructFuncOfLeaf(columnIndex int16, node Node) (int16, reconstructFunc) {
+	typ := node.Type()
 	return columnIndex + 1, func(value reflect.Value, _ levels, row Row) (Row, error) {
 		if !row.startsWith(columnIndex) {
 			return row, fmt.Errorf("no values found in parquet row for column %d", columnIndex)
 		}
-		return row[1:], assignValue(value, row[0])
+
+		var err error
+		src := row[0]
+
+		if src.IsNull() {
+			value.Set(reflect.Zero(value.Type()))
```

Comment:
I wonder if this would be better implemented by being moved to each `AssignValue` method: in most cases we go through the specialized code path (e.g. `SetInt`) where a null parquet value would result in the zero value and have the same effect but would be more efficient since the code wouldn't have to instantiate the zero `reflect.Value`.

We are also bypassing type checking in this branch, where we don't validate that the destination Go value was actually assignable from parquet values of the parquet type.

Reply:
🤔 That's a good idea. Looks like the only function on `Value` that doesn't appreciate being called on a null value is `ByteArray()` (and `Int96()` by extension). All the others return a zero value so the implementation can be ignorant of null values. How does c6efc31 look?

Reply:
That's surprising, on a null value the ptr+len will be nil and zero 
https://github.com/segmentio/parquet-go/blob/main/value.go#L405 which should still be a valid byte slice?
```go
func (v Value) ByteArray() []byte { return unsafe.Slice(v.ptr, int(v.u64)) }
```

Reply:
Let's make a special case for Int96 only then 👍

Reply:
Done in 3381746